CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -0,0 +1,172 @@
+        /// This is the amount that all players must match to stay in the game.
+        /// </summary>
+        private int _currentBet;
+        
+        /// <summary>
+        /// The Small Blind for the game.
+        /// </summary>
+        private int _smallBlind;
+        
+        /// <summary>
+        /// The Big Blind for the game.
+        /// </summary>
+        private int _bigBlind;
+        
+        /// <summary>
+        /// The Ante for the game.
+        /// This is the amount that all players must pay to play the game.
+        /// </summary>
+        private int _ante;
+        
+        /// <summary>
+        /// A dictionary that tracks which player has what role.
+        /// </summary>
+        private Dictionary<PlayerRole, int> _players { get; set; }
+        
+        private List<CardHand> _playerHands { get; set; }
+
+        public TexasHoldEmManager()
+        {
+            _cardManager = new CardManager();
+            _userHand = new UserHand();
+            _gameOver = false;
+            _chipManager = new ChipManager();
+            _mainPot = 0;
+            _sidePot = 0;
+            _currentBet = 0;
+            _smallBlind = 0;
+            _bigBlind = 0;
+            _ante = 0;
+
+            // Initialize the player hands. The first player is the user.
+            for (int i = 0; i < Constants.AI_PLAYER_COUNT; i++)
+            {
+                if (i == 0)
+                {
+                    _playerHands[i] = new UserHand();
+                }
+                else
+                {
+                    _playerHands[i] = new PokerHand();
+                }
+            }
+            
+            // Initialize the player roles. The user starts as the dealer. 
+            _players = new Dictionary<PlayerRole, int>
+            {
+                {PlayerRole.DEALER, 0},
+                {PlayerRole.SMALL_BLIND, 1},
+                {PlayerRole.BIG_BLIND, 2}
+            };
+        }
+
+        public void StartNewGame()
+        {
+            _playerHands.Clear();
+            _gameOver = false;
+            
+            
+        }
+
+        /// <summary>
+        /// Deals cards to all players in the game starting with the player with the small blind and ending with the dealer.
+        /// </summary>
+        public void DealCards()
+        {
+            // The index of the current player with the small blind.
+            // Initialized to avoid a null reference exception in line 171.
+            int currentIndex = _players[PlayerRole.SMALL_BLIND];
+
+            // Iterates through the players in a circular fashion, starting with the player with the small blind.
+            for (int i = 0; i < _players.Count; i++) 
+            {
+                int playerIndex = (currentIndex + i) % _players.Count;
+                
+                // Deals two cards to each player.
+                _playerHands[playerIndex].AddCard(_cardManager.DrawCard());
```

Comment:
This is incorrect implementation. you don't deal both cards at the same time, you give each player one card and then go around a second time and give each player a second card.

Mostly a technicality because the speed of computers means this is instant, but I think we are likely going to want to implement some sort of pause in the future to make the game look better when updating the screen, so we are going to want to have the implementation correct.